fdbclient/include/fdbclient/BlobCipher.h

```diff
@@ -181,6 +192,186 @@ struct hash<BlobCipherDetails> {
+// header} payload. Scheme is geared towards optimizing cost due to crypto-secure auth-token generation,
+// however, on decryption client needs to be read 'header' + 'encrypted-buffer' to validate the 'auth-token'.
+// The scheme is ideal for usecases where payload represented by the encryptionHeader is not large and it is
+// desirable to minimize CPU/latency penalty due to crypto-secure ops, such as: CommitProxies encrypted inline
+// transactions, StorageServer encrypting pages etc.
+// SOMEDAY: Another potential scheme could be 'MultiAuthTokenMode': Scheme generates separate authTokens
+// for 'encrypted buffer' & 'encryption-header'. The scheme is ideal where payload represented by
+// encryptionHeader is large enough such that it is desirable to optimize cost of upfront reading full
+// 'encrypted buffer', compared to reading only encryptionHeader and ensuring its sanity; for instance:
+// backup-files.
+
+template <uint32_t AuthTokenSize>
+struct AesCtrWithAuthV1 {
+	// Serializable fields
+
+	// Text cipher encryption information
+	BlobCipherDetails cipherTextDetails;
+	// Header cipher encryption information
+	BlobCipherDetails cipherHeaderDetails;
+	// Initialization vector
+	uint8_t iv[AES_256_IV_LENGTH];
+	// Authentication token
+	uint8_t authToken[AuthTokenSize];
+
+	AesCtrWithAuthV1() {}
+	AesCtrWithAuthV1(const BlobCipherDetails& textDetails,
+	                 const BlobCipherDetails& headerDetails,
+	                 const uint8_t* ivBuf,
+	                 const int ivLen,
+	                 Arena& arena)
+	  : cipherTextDetails(textDetails), cipherHeaderDetails(headerDetails) {
+		ASSERT_EQ(ivLen, AES_256_IV_LENGTH);
+		memcpy(&iv[0], ivBuf, ivLen);
+		memset(&authToken[0], 0, AuthTokenSize);
+	}
+
+	bool operator==(const AesCtrWithAuthV1<AuthTokenSize>& o) const {
+		return cipherHeaderDetails == o.cipherHeaderDetails && cipherTextDetails == o.cipherTextDetails &&
+		       memcmp(&iv[0], &o.iv[0], AES_256_IV_LENGTH) == 0 &&
+		       memcmp(&authToken[0], &o.authToken[0], AuthTokenSize) == 0;
+	}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		ar.serializeBytes(this, sizeof(AesCtrWithAuthV1<AuthTokenSize>));
+	}
+};
+
+struct AesCtrNoAuthV1 {
+	// Serializable fields
+
+	// Text cipher encryption information
+	BlobCipherDetails cipherTextDetails;
+	// Initialization vector
+	uint8_t iv[AES_256_IV_LENGTH];
+
+	AesCtrNoAuthV1() {}
+	AesCtrNoAuthV1(const BlobCipherDetails& textDetails, const uint8_t* ivBuf, const int ivLen, Arena& arena)
+	  : cipherTextDetails(textDetails) {
+		ASSERT_EQ(ivLen, AES_256_IV_LENGTH);
+		memcpy(&iv[0], ivBuf, ivLen);
+	}
+
+	bool operator==(const AesCtrNoAuthV1& o) const {
+		return cipherTextDetails == o.cipherTextDetails && memcmp(&iv[0], &o.iv[0], AES_256_IV_LENGTH) == 0;
+	}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		ar.serializeBytes(this, sizeof(AesCtrNoAuthV1));
+	}
+};
+
+struct ParsedEncryptHeaderDetails {
```

Comment:
I think this is no longer needed?

Reply:
I would keep this around, as from the StringRef header one needs to read/extract the right structs, in test it is used. So if no strong opinions I would just keep it.

Reply:
Thinking more may be it would be more work to keep this routine updated in the longer run, will remove it (let me think a bit more...)
Thanks